packages/instant/src/components/animations/slide_up_and_down_animation.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+
+import { keyframes, styled } from '../../style/theme';
+
+const slideKeyframeGenerator = (fromY: string, toY: string) => keyframes`
+    from {
+        position: relative;
+        top: ${fromY}
+    }
+
+    to {
+        position: relative;
+        top: ${toY}
+    }
+`;
+
+export interface SlideAnimationProps {
+    keyframes: string;
+    animationType: string;
+    animationDirection?: string;
+}
+export const SlideAnimation =
+    styled.div <
+    SlideAnimationProps >
+    `
+    animation-name: ${props => props.keyframes};
+    animation-duration: 0.3s;
+    animation-timing-function: ${props => props.animationType};
+    animation-delay: 0s;
+    animation-iteration-count: 1;
+    animation-fill-mode: ${props => props.animationDirection || 'none'};
+    position: relative;
+    z-index: -1;
+`;
+
+export const SlideUpAnimationComponent: React.StatelessComponent<{ downY: string }> = props => (
+    <SlideAnimation animationType="ease-in" keyframes={slideKeyframeGenerator(props.downY, '0px')}>
+        {props.children}
+    </SlideAnimation>
+);
+
+export const SlideDownAnimationComponent: React.StatelessComponent<{ downY: string }> = props => (
+    <SlideAnimation
+        animationDirection="forwards"
+        animationType="cubic-bezier(0.25, 0.1, 0.25, 1)"
+        keyframes={slideKeyframeGenerator('0px', props.downY)}
+    >
+        {props.children}
+    </SlideAnimation>
+);
+
+export interface SlideUpAndDownAnimationProps {
+    delayMs: number;
+    downY: string;
+}
+
+interface SlideUpAndDownState {
+    slideState: 'up' | 'down';
+}
+
+export class SlideUpAndDownAnimationComponent extends React.Component<
+    SlideUpAndDownAnimationProps,
+    SlideUpAndDownState
+> {
+    private _timeoutNumber: number | undefined;
+
+    constructor(props: SlideUpAndDownAnimationProps) {
+        super(props);
+        this._timeoutNumber = undefined;
+        this.state = {
+            slideState: 'up',
+        };
+    }
```

Comment:
this can just be `state` without the constructor. The constructor is only really necessary if you're accessing props. 

Also to type `props` as `SlideUpAndDownAnimation` is not technically correct but should work and I'm not sure what to think about it. Internally the react types take your `SlideUpAndDownAnimationProps` and adds the children prop such that `props` is actually `SlideUpAndDownAnimationProps & { children?: React.ReactNode }`

Reply:
Good call on not needing the constructor to set `state`, I didn't know this.  Although as per slack convo, we had to change `slideState` to be an `Enum` to get typesafety working.

I think you're right that it's good to be explicit about `children` being a prop, will add

Reply:
Oh my point was that you're unnecessarily restricting the `props` object to something without children. But was not really an actionable point :) I think the action there would be to leave `props` untyped (if possible?) and let ts infer the type.